subgraphs/ribbon-finance/src/common/initializers.ts

```diff
@@ -0,0 +1,507 @@
+import {
+  Account,
+  FinancialsDailySnapshot,
+  RewardToken,
+  Token,
+  UsageMetricsDailySnapshot,
+  UsageMetricsHourlySnapshot,
+  YieldAggregator,
+  _Auction,
+  _LiquidityGauge,
+  Vault as VaultStore,
+  VaultHourlySnapshot,
+  VaultDailySnapshot,
+  VaultFee,
+  _SwapOffer,
+} from "../../generated/schema";
+import * as utils from "./utils";
+import * as constants from "./constants";
+import { getUsdPricePerToken } from "../prices";
+import { Address, ethereum, BigInt } from "@graphprotocol/graph-ts";
+import { ERC20 as ERC20Contract } from "../../generated/ETHCallV2/ERC20";
+import { LiquidityGauge as LiquidityGaugeTemplate } from "../../generated/templates";
+import { LiquidityGaugeV5 as GaugeContract } from "../../generated/rETHThetaGauge/LiquidityGaugeV5";
+import { RibbonThetaVaultWithSwap as VaultContract } from "../../generated/ETHCallV2/RibbonThetaVaultWithSwap";
+
+export function getOrCreateToken(
+  address: Address,
+  block: ethereum.Block,
+  vault: Address = constants.NULL.TYPE_ADDRESS,
+  isOutputToken: bool = false
+): Token {
+  let token = Token.load(address.toHexString());
+
+  if (!token) {
+    token = new Token(address.toHexString());
+
+    if (!isOutputToken) {
+      const contract = ERC20Contract.bind(address);
+
+      token.name = utils.readValue<string>(contract.try_name(), "");
+      token.symbol = utils.readValue<string>(contract.try_symbol(), "");
+      token.decimals = utils.readValue<i32>(
+        contract.try_decimals(),
+        constants.DEFAULT_DECIMALS.toI32() as u8
+      );
+      token._isOutputToken = false;
+      token._vaultId = vault.toHexString();
+
+      token.lastPriceUSD = getUsdPricePerToken(address).usdPrice;
+      token.lastPriceBlockNumber = block.number;
+    }
+
+    if (isOutputToken) {
+      const contract = VaultContract.bind(address);
+      token.name = utils.readValue<string>(contract.try_name(), "");
+      token.symbol = utils.readValue<string>(contract.try_symbol(), "");
+      token.decimals = utils.readValue<i32>(
+        contract.try_decimals(),
+        constants.DEFAULT_DECIMALS.toI32() as u8
+      );
+      token._isOutputToken = true;
+      token._vaultId = vault.toHexString();
+
+      token.lastPriceUSD = utils.getOutputTokenPriceUSD(vault, block);
+      token.lastPriceBlockNumber = block.number;
+    }
+    token.save();
+  }
+
+  if (
```

Comment:
I removed the condition for updating the price once per hour, now I am updating the price whenever getOrCreateToken is called and changed the prices call in updateVaultTVL to getOrCreateToken so that it also updates the price.